samples/extensions/open_cl_interop/open_cl_interop.h

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2021, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+#include "rendering/render_pipeline.h"
+#include "scene_graph/components/camera.h"
+
+#include <android/hardware_buffer.h>
+#include <android/hardware_buffer_jni.h>

```

Comment:
these includes can be moved into an appropriate if macro check to ensure on Android.

Also, use forward declare of struct AHardwareBuffer; in the header then include these in the CPP to allow for less tightly coupling of header files.